frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -508,19 +509,20 @@ export default ({
               return false
             }
           }
-        })
-        uploadAttachments().then((isUploaded) => {
-          const removeTemporaryMessage = () => {
-            // NOTE: remove temporary message which is created before uploading attachments
-            if (temporaryMessage) {
-              const msgIndex = findMessageIdx(temporaryMessage.hash, this.messageState.contract.messages)
-              this.messageState.contract.messages.splice(msgIndex, 1)
-            }
-          }
+        }).then(async () => {
+          const isUploaded = await uploadAttachments()
           if (isUploaded) {
+            const removeTemporaryMessage = () => {
+              // NOTE: remove temporary message which is created before uploading attachments
+              if (temporaryMessage) {
+                const msgIndex = findMessageIdx(temporaryMessage.hash, this.messageState.contract.messages)
+                this.messageState.contract.messages.splice(msgIndex, 1)
+              }
+            }
             sendMessage(removeTemporaryMessage)
           } else {
-            removeTemporaryMessage()
+            temporaryMessage.pending = false
```

Comment:
This line doesn't make sense to me. The message should still be pending if the files failed to upload (see description of message ordering behavior in #1845).